### DISCUSSION TITLE
I would like to create a service as an extension

### DISCUSSION THREAD

**Discussion by diegoleme** [directus/directus#12470] [Feature Requests]
Today I have some integrations with third-party systems.

In each extension I use one of these integrations I have to instantiate a new service for that integration. 

Example: Every time an item is created, I send a request to an external service that I need to authenticate. E autentico nesse serviço em todas as vezes.

I would like to have an instance of this service shared with all my extensions!

I believe we could have an event on a start hook. something like:

I'm suggesting a new type of extension called `service`.

And that it is accessed in hooks and endpoints along with intern services.

It makes sense?

**Comment by rijkvanzanten** [comment]
I don't think that has to be a new extension necessarily! You should be able to just really on Node's built-in module resolution, by doing something like:

```js
// my-shared-service.js

let service;

function getService() {
  if (!service) service = new MyExternalService();
  return service;
}

module.exports = { getService };

// And then in extensions:
// hooks/my-hook/index.js
const { getService } = require('../my-shared-service');

const sharedService = getService();
```

**Reply by diegoleme** [reply]
I liked the solution! Simple and it should work.

Thank you @rijkvanzanten

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
